### PR TITLE
Added Email index and show view 

### PR DESCRIPTION
**Pull request by nikhilgupta1211**

### CODE REVIEW

**Comment by bgeuken** [comment]
The breadcrumbs are missing;-)

File: app/models/email_event.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class EmailEvent < ActiveRecord::Base
```
Comment (bgeuken) [inline app/models/email_event.rb]: Please rename it to EventEmail

File: db/migrate/20170620170817_createemailevents.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Createemailevents < ActiveRecord::Migration
```
Comment (bgeuken) [inline db/migrate/20170620170817_createemailevents.rb]: Classes should be written in CamelCase

File: app/controllers/email_events_controller.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  def create
+    @email_event = EmailEvent.new(create_params)
+    @email_event.event_id = @event.id
+
+    if @email_event.save
+      @email_event[:to].split(',').each do |e|
+        user = User.find_by(email: e)
+        if user.nil?
+          flash[:error] = 'Mail did not delivered to ' + e + ', because this email is not registered with TSP'
```
Comment (bgeuken) [inline app/controllers/email_events_controller.rb]: I would rephrase the first part to "Email was not delivered to". And maybe put the second part into a separate sentence, like "This email is not registered with TSP."

File: app/controllers/email_events_controller.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    else
+      render :new
+    end
+  end
+
+  def show
+    @email_event = EmailEvent.find(params[:id])
+  end
+
+  def event
```
Comment (bgeuken) [inline app/controllers/email_events_controller.rb]: What is this action used for?

File: app/controllers/events_controller.rb [hunk 5]
```diff
@@ -4,22 +4,6 @@ class EventsController < InheritedResources::Base
   skip_load_and_authorize_resource only: [:index, :show]
```
Comment (bgeuken) [inline app/controllers/events_controller.rb]: This commit and the previous one seem to belong together (Moving files and code). In such a case it makes sense to have the changes in one commit.
Please squash this commit into the previous one.

File: spec/models/ability_for_tsp_spec.rb [hunk 6]
```diff
@@ -21,10 +21,6 @@
     it { should be_able_to(:update, events(:party)) }
     it { should_not be_able_to(:destroy, events(:yavin_hackaton)) }
     it { should_not be_able_to(:destroy, events(:party)) }
-    it { should be_able_to(:email, events(:yavin_hackaton)) }
-    it { should be_able_to(:email, events(:party)) }
-    it { should be_able_to(:email_event, events(:yavin_hackaton)) }
-    it { should be_able_to(:email_event, events(:party)) }
```
Comment (bgeuken) [inline spec/models/ability_for_tsp_spec.rb]: I guess we should add tests for managing event mails instead

File: app/models/event_email.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+class EventEmail < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :event
+
+  validates :to, presence: true
+  validates :subject, presence: true, length: { maximum: 150 }
+  validates :body, presence: true
```
Comment (bgeuken) [inline app/models/event_email.rb]: We are using [shoulda matchers](http://www.rubydoc.info/github/thoughtbot/shoulda-matchers/Shoulda/Matchers) which provides some nice methods for testing validations.
Please add them to the event mail tests.

File: app/assets/stylesheets/event_email.css.less [hunk 8]
```diff
@@ -0,0 +1,17 @@
+.event_email_show {
+  .well {
+    word-wrap: break-word;
+  }
+
+  .labl {
```
Comment (bgeuken) [inline app/assets/stylesheets/event_email.css.less]: Please rename  this one. If it's a label for some specific group of html elements call it *-label:-)

File: app/models/ability.rb [hunk 9]
```diff
@@ -1,9 +1,11 @@
 #
 # CanCan permissions
 #
+# rubocop:disable ClassLength
```
Comment (bgeuken) [inline app/models/ability.rb]: Since this is meant for the whole file, it's better to add this as an exception to the rubocp.yml file.

File: app/models/ability.rb [hunk 10]
```diff
@@ -1,9 +1,11 @@
...
 class Ability
   include CanCan::Ability
 
+  # rubocop:disable MethodLength,AbcSize
```
Comment (bgeuken) [inline app/models/ability.rb]: You need to enable this cop again. Otherwise it stays disabled.

File: app/views/event_emails/show.html.haml [hunk 11]
```diff
@@ -0,0 +1,9 @@
+.col-md-11.event_email_show
+	%h3 To:
+	.labl
+		- @event_email.to.split(',').each do |t|
+			%span.label.label-info #{t}
+	%h3 Subject:
+	.well.well-sm #{@event_email.subject}
+	%h3 Body:
+	.well #{@event_email.body}
```
Comment (bgeuken) [inline app/views/event_emails/show.html.haml]: Please fix the indentation here.

File: spec/models/event_email_spec.rb [hunk 12]
```diff
@@ -0,0 +1,10 @@
+require 'spec_helper'
+
+describe EventEmail do
+  fixtures :all
+
+  it { should validate_presence_of :to }
+  it { should validate_presence_of :subject }
+  it { should validate_length_of(:subject).is_at_most(150) }
+  it { should validate_presence_of :body }
```
Comment (bgeuken) [inline spec/models/event_email_spec.rb]: Ah, you already did this :+1:

File: app/models/event_email.rb (hunk 7 above)
Comment (bgeuken) [inline app/models/event_email.rb]: Nevermind. I just saw your other commit

File: app/models/ability.rb (hunk 9 above)
Comment (nikhilgupta1211) [inline app/models/ability.rb]: But if we add the file in rubocop.ym no other cops will also work on the file

Comment (bgeuken) [inline app/models/ability.rb]: You can exclude the cop for certain files with something like this:

```
MyCop:
  Exclude:                                                                                                                                                                
    - 'foo/bar/test.rb'
```

File: .rubocop.yml [hunk 13]
```diff
@@ -1,6 +1,18 @@
 # Inherit style from another configuration
 inherit_from: .rubocop_todo.yml
 
+ClassLength:
+   Exclude:
+    - 'app/models/ability.rb'
+
+MethodLength: 
+ Exclude:
+    - 'app/models/ability.rb'
```
Comment (bgeuken) [inline .rubocop.yml]: Tried it on my machine and it seems that only the initialize method offends that cop. Please just disable that cop.

File: .rubocop.yml [hunk 14]
```diff
@@ -1,6 +1,18 @@
...
+
+AbcSize:
+  Exclude:
+    - 'app/models/ability.rb'
```
Comment (bgeuken) [inline .rubocop.yml]: Same here. Please just disable the cop for the single method, like you did before. Just re-enable the cop at the end of the method.

**Comment by bgeuken** [comment]
There is a bug in the breadcrumbs. When I create a email and then click on the event breadcrumb it directs me to a nonexistant page.
It looks like it uses the email id as event id.

**Comment by bgeuken** [comment]
Another issue is that newlines I add to the email body get lost. So when I view a sent email it's a big blob of text, which makes it hard to read.
